File: JavaCivitas/src/civitas/SorpresaIrCarcel.java
package civitas;

import java.util.ArrayList;

public class SorpresaIrCarcel extends Sorpresa{

    private Tablero tablero;

    SorpresaIrCarcel(Tablero tablero_a, String texto){

        super(texto);
        tablero =  tablero_a;

    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){

        if (jugadorCorrecto(actual, todos) && todos.get(actual) != null){

            informe(actual, todos);
            todos.get(actual).encarcelar(tablero.getCarcel());

        }

    }

}

File: RubyCivitas/lib/sorpresa_ircarcel.rb
module Civitas
  class Sorpresa_ircarcel < Sorpresa
    
    def initialize(tablero)
      
      @texto = "Sorpresa para ir a la carcel"
      @tablero = tablero
      
    end
    
    def aplicar_a_jugador(actual, todos)

      if jugador_correcto(actual, todos)

        informe(actual, todos)
        todos[actual].encarcelar(@tablero.num_casilla_carcel)

      end

    end
    
    
  end
end